honeybee_3dm/config.py

```diff
@@ -0,0 +1,196 @@
+    """List of acceptable string names for Honeybee Face object types."""
+    door = 'door'
+    shade = 'shade'
+    aperture = 'aperture'
+
+
+class FaceType(str, enum.Enum):
+    """List of acceptable string names for Honeybee face types."""
+    wall = 'wall'
+    roof = 'roof'
+    floor = 'floor'
+    airwall = 'airwall'
+
+
+class GridSettings(BaseModel):
+    """Config for the grid-settings."""
+
+    grid_size: float = Field(
+        1.0,
+        gt=0,
+        description='Grid spacing to control grid density.'
+    )
+
+    grid_offset: float = Field(
+        0.0,
+        description='Offset to move grid away from the parent face by certain distance.'
+    )
+
+
+class LayerConfig(BaseModel):
+    """Config for layer keys."""
+
+    exclude_from_rad: Optional[bool] = Field(
+        description='Boolean to indicate if radiance material is applied to this layer.'
+            ' This is useful for layers on which grid objects are saved.'
+    )
+
+    include_child_layers: Optional[bool] = Field(
+        True,
+        description='Boolean to indicate if objects from child layers are to be imported.'
+    )
+
+    radiance_material: Optional[str] = Field(
+        description='The name of radiance modifier.'
+    )
+
+    grid_settings: Optional[GridSettings] = Field(
+        description='Grid settings for the layer.'
+    )
+
+    honeybee_face_object: Optional[FaceObject] = Field(
+        description='A text string for FaceObject to create either a Honeybee Shade,'
+        ' a Honeybee Door, or a Honeybee Aperture object.'
+    )
+
+    honeybee_face_type: Optional[FaceType] = Field(
+        description='A text string for FaceType to create either a Honeybee Wall,'
+        ' a Honeybee Floor, a Honeybee Roof/Ceiling, or a Honeybee Airwall.'
+    )
+
+
+class Config(BaseModel):
+    """Config file."""
+
+    sources: Dict[str, str] = Field(
+        None,
+        description='Path to input sources. Currently, the config only supports'
+            ' "radiance_material" to indicate the path to the radiance.mat file.'
+        )
+
+    layers: Dict[str, LayerConfig] = Field(
+        ...,
+        description='A dictionary to indicate the config for each layer in rhino 3dm'
+            ' file. The key must be a Rhino layer name and the values must be a'
+            ' LayerConfig.'
+        )
+
+    @validator('sources')
+    def check_sources(cls, v):
+        sources = v
+        if sources:
+            sources = list(v.keys())
+            if len(sources) > 1:
+                raise ValueError('sources can only have one key.')
+
+            if sources[0] != 'radiance_material':
+                raise ValueError(
+                        f'invalid sources key: {sources[0]}.'
+                        'key must be radiance_material.'
+                )
+        return v
+
+    @validator('layers')
+    def check_rad(cls, v, values):
+        config_layers = v
+        sources = values['sources']
```

Comment:
You have to do a quick check here to ensure the 'sources' key is not None before trying to get the value.

Reply:
The value will be none if the path is not provided and I am dealing with the None value.